UnityProject/Assets/Scripts/Transform/CustomNetTransform.cs

```diff
@@ -542,6 +545,22 @@ private void AdminRespawn()
 	{
 		PlayerManager.PlayerScript.playerNetworkActions.CmdAdminRespawn(gameObject);
 	}
+
+	// Checks if the object is occupiable and update occupant position if it's occupied (ex: a chair)
+	private void UpdateOccupant()
+	{
+		// Since we are at the transform level, we go up one level to get the parent
+		OccupiableDirectionalSprite occupiableDirectionalSprite = GetComponentInParent<OccupiableDirectionalSprite>();
+		if ((occupiableDirectionalSprite != null) && (occupiableDirectionalSprite.Occupant > 0))
+		{
+			NetworkIdentity networkIdentity = NetworkIdentity.spawned[occupiableDirectionalSprite.Occupant];
+			if (networkIdentity != null && networkIdentity.TryGetComponent<PlayerMove>(out var playerMove))
+			{
+				//sync position to ensure they buckle to the correct spot
+				playerMove.PlayerScript.PlayerSync.SetPosition(occupiableDirectionalSprite.gameObject.TileWorldPosition().To3Int());
```

Comment:
```suggestion
				playerMove.PlayerScript.PlayerSync.SetPosition(registerTile.WorldPosition);
```

TileWorldPosition can cause GC, this should avoid that. Getting registered position through a cached register tile is the way to avoid TileWorldPosition() and other similar extension methods that cause GC.

Reply:
So... Basically, a "registerTile" is the tile (with coordinates) an object is in?

Reply:
Not quite, registerTile is a component that lives on each object (not on each tile) which TRACKS which position and matrix the object is in. It's a bit of an unfocused component TBH and it's quite old.